packages/netlify-cms-backend-github/src/GraphQLAPI.js

```diff
@@ -0,0 +1,459 @@
+import { ApolloClient } from 'apollo-client';
+import {
+  InMemoryCache,
+  defaultDataIdFromObject,
+  IntrospectionFragmentMatcher,
+} from 'apollo-cache-inmemory';
+import { createHttpLink } from 'apollo-link-http';
+import { setContext } from 'apollo-link-context';
+import { APIError, EditorialWorkflowError } from 'netlify-cms-lib-util';
+import introspectionQueryResultData from './fragmentTypes';
+import API from './API';
+import * as queries from './queries';
+import * as mutations from './mutations';
+
+const NO_CACHE = 'no-cache';
+const CACHE_FIRST = 'cache-first';
+
+const TREE_ENTRY_TYPE_TO_MODE = {
+  blob: '100644',
+  tree: '040000',
+  commit: '160000',
+};
+
+const fragmentMatcher = new IntrospectionFragmentMatcher({
```

Comment:
see https://www.apollographql.com/docs/react/advanced/fragments/#fragments-on-unions-and-interfaces